### ISSUE TITLE
Instrumentation tests fail when navigating to dynamic feature

### ISSUE THREAD

**Issue by RafaO** [android/app-bundle-samples#4]
Hey!

I have a problem when trying to write a test that performs a navigation action into an activity hosted in a dynamic feature module.

This navigation fails when executed through the gradle task, e.g the task `connectedAndroidTest` fails, while running with the adb shell command
` adb shell am instrument -w -r com.google.android.samples.dynamicfeatures.ondemand.test/androidx.test.runner.AndroidJUnitRunner` succeeds. 

I modified a little bit the code to perform the navigation inside a dynamic feature when clicking a button in the main activity. Here is the code I am using:
```
    private fun intentTo(className: String): Intent =
            Intent(Intent.ACTION_VIEW).setClassName(BuildConfig.APPLICATION_ID, className)

    private fun String.loadIntentOrNull(): Intent? =
            try {
                Class.forName(this).run { intentTo(this@loadIntentOrNull) }
            } catch (e: ClassNotFoundException) {
                null
            }

    private val clickListener by lazy {
        View.OnClickListener {
            when (it.id) {
                R.id.btn_load_kotlin -> {
                    val intent = KOTLIN_SAMPLE_CLASSNAME.loadIntentOrNull()
                    startActivity(intent)
                }
                R.id.btn_load_java -> loadAndLaunchModule(moduleJava)
                R.id.btn_load_assets -> loadAndLaunchModule(moduleAssets)
                R.id.btn_load_native -> loadAndLaunchModule(moduleNative)
                R.id.btn_install_all_now -> installAllFeaturesNow()
                R.id.btn_install_all_deferred -> installAllFeaturesDeferred()
                R.id.btn_request_uninstall -> requestUninstall()
                R.id.btn_instant_dynamic_feature_split_install -> loadAndLaunchModule(instantModule)
                R.id.btn_instant_dynamic_feature_url_load -> openUrl(instantModuleUrl)
                R.id.lang_en -> loadAndSwitchLanguage(LANG_EN)
                R.id.lang_pl -> loadAndSwitchLanguage(LANG_PL)
            }
        }
    }
```

When executing the app, it is able to navigate without problems but the instrumentation tests aren't, because the intent is null since it appears it cannot find the activity. Here is the test: 
```
    @Test
    fun launchMainActivity() {
        onView(withId(R.id.btn_load_kotlin)).perform(ViewActions.click())
    }
```

I am interested in the test results reports that are only generated with the gradle task and not through the adb shell command.

Thank you in advance! :)

**Comment by keyboardsurfer** [comment]
From Studio there's a bug we're [currently investigating](https://issuetracker.google.com/142043473). You should be able to run the tests from command line using `./gradlew connectedAndroidTest --no-parallel`.

**Comment by keyboardsurfer** [comment]
Resolved with AndroidStudio 4.0 Beta 04 release.
